docs/specs/lcb-signals-and-events.md

```diff
@@ -0,0 +1,222 @@
+
+2. This permits simple implementation of syntactic sugar for frequently-needed event data.  For example:
+
+       the mouse position of the event
+
+   can easily be implemented as:
+
+       the event["mouse position"]
+
+
+3. It might be useful to provide `the event target is me` (or equivalent syntax) to test whether an event is targeted at the handling control or one of its children.
+
+
+### Event handlers
+
+All event handlers (for enter, dispatch and leave phases) must conform to:
+
+    public type EventHandler() returns any
+
+The return value of all event handlers is discarded.
+
+### Enter phase
+
+During the enter phase, **pre-event** handlers are run.  They are run from "outside-in"; a parent widget's pre-event handler runs before the pre-event handler of its child widgets.
+
+A pre-event handler is declared using a `pre event handler` declaration:
+
+    pre event handler for <name:id> is <handler:constexpr>
+
+By default, all pre-event handlers declared by the list of interested controls are run.  There is no need to "pass" the event onwards [1].
+
+A control can use the pre-event handler to **capture** an event:
+
+    capture event
+
+The `capture event` statement makes the capturing control becomes the "target" for the event; all its children are removed from the list of interested controls [2,3].
+
+**Notes:**
+
+1. As per the `mouseEnter`/`mouseLeave`, implicit passing reflects the expectation that pre- and post-event handlers are usually be used to implement behaviour that complements rather than conflicts with that of other interested controls.
+
+2. This allows composed widgets to "hide" events from their children.
+
+3. Since `capture event` causes the capturing control to become the target, the capturing control can't do anything that requires knowledge of the "original" target in its on-event handler unless it saves `the event target` in the pre-event handler before `capture event`.  On the other hand, there's not much point in having separate pre-event and on-event handlers when capturing because they're guaranteed to be run immediately one after another.
+
+### Dispatch phase
+
+During the dispatch phase, **on-event** handlers are run.  They are run from "inside-out", starting from the target control.  A parent widget's on-event handler runs after its child widgets' on-event handlers.
+
+An on-event handler is declared using an `event handler` declaration [1]:
+
+    [on] event handler for <name:id> is <handler:constexpr>
+
+By default, if an on-event handler is found for an event, then it **blocks** further on-event handler execution for the event [2].
+
+An on-event handler can use the `pass event` statement to prevent the default-blocking behaviour.
+
+The `pass event` statement does not return.
+
```

Comment:
@montegoulding: Widgets are always entirely opaque to LiveCode Script - they are black-boxes. So 'the target' will always be the widget and never its children. As an analog, think of the engine combo-box, that contains a field and the 'popup' button. You can't access the field directly, instead the control provides a means to access the field in the ways that make sense for that control.